### PR TITLE
add rollback service

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
Unified RollbackService

### CODE REVIEW

File: notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import com.github.kittinunf.result.flatMap
+import com.github.kittinunf.result.map
+import jp.co.soramitsu.iroha.java.Transaction
+import mu.KLogging
+import java.math.BigDecimal
+
+/**
+ * Class that returns assets in Iroha in case of failed withdrawal
+ */
+class RollbackService(val withdrawalConsumer: IrohaConsumer) {
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt]: I'd rather make it `private`

**Review by Mingela — APPROVED** [review]

File: notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    .transferAsset(
+                        withdrawalDetails.destinationAddress,
+                        withdrawalDetails.srcAccountId,
+                        withdrawalDetails.withdrawalAssetId,
+                        "$ROLLBACK_DESCRIPTION. $reason".take(64).toLowerCase(),
+                        withdrawalDetails.withdrawalAmount
+                    )
+
+                // if fee > 0
+                if (withdrawalDetails.feeAmount.compareTo(BigDecimal.ZERO) == 1)
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt]: Just make it `withdrawalDetails.feeAmount>BigDecimal.ZERO`. Kotlin can handle it.

File: notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                transaction.build()
+            }.flatMap { transaction ->
+                withdrawalConsumer.send(transaction)
+            }
+    }
+
+    /**
+     * Logger
+     */
+    companion object : KLogging()
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt]: Not used

File: notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    fun rollback(
+        withdrawalDetails: WithdrawalFinalizationDetails,
+        reason: String
+    ): Result<String, Exception> {
+        return withdrawalConsumer.getConsumerQuorum()
+            .map { quorum ->
+                val transaction = Transaction
+                    .builder(withdrawalConsumer.creator, withdrawalDetails.withdrawalTime)
+                    .setQuorum(quorum)
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/service/RollbackService.kt]: We must set time as well. I use MST to rollback.

**Review by dolgopolovwork — CHANGES_REQUESTED** [review]
Please add time

**Review by dolgopolovwork — APPROVED** [review]

**Pull request merged**